tests/test_common_gpu.py

```diff
@@ -316,3 +317,40 @@ def test_print_4bit_expected(self):
 
         self.assertEqual(trainable_params, EXPECTED_TRAINABLE_PARAMS)
         self.assertEqual(all_params, EXPECTED_ALL_PARAMS)
+
+    @require_torch_gpu
+    @pytest.mark.single_gpu_tests
+    @require_bitsandbytes
+    def test_modules_to_save_grad(self):
```

Comment:
Do we want to throw in the `device_map="auto"` option too?

Reply:
`load_in_4bit` should automatically set `device_map` to the correct value: https://github.com/huggingface/transformers/blob/a0042379269bea9182c1f87e6b2eee4ba4c8cce8/src/transformers/modeling_utils.py#L2318

Reply:
Okay. I was just thinking about the users reporting that the issue also occurs with `device_map="auto"` and without quantization, so we could cover that too. But it's not super important.